src/org/stringtemplate/v4/STGroup.java

```diff
@@ -80,9 +80,11 @@
 		TEMPLATE_FILE_EXTENSION = ".st";
 	}
 
-    /** When we use key as a value in a dictionary, this is how we signify. */
-    public static final String DICT_KEY = "key";
-    public static final String DEFAULT_KEY = "default";
+    /** When we use key as a value in a dictionary, this is how we signify.
+     *  Magic suffix is used to distinguish "default" string literal and "default" key
+     *  End part is the date of this suffix adding. */
+    public static final String DICT_KEY = "key_st4_magic_suffix_2016/01/10";
+    public static final String DEFAULT_KEY = "default_st4_magic_suffix_2016/01/10";
```

Comment:
I tried this, but unfortunately is does not solve the problem. See [this test on gist](
https://gist.github.com/KvanTTT/6cac1d75061995e96a71).
Escaped chars are allowed in keys (actually key can be any string due to escaping).